web/src/views_util.js

```diff
@@ -43,6 +42,9 @@ export function show(opts) {
     search.clear_search_form();
     opts.complete_rerender();
 
+    // This has to happen after resetting the current filter.
+    pm_list.handle_message_view_deactivated();
+
```

Comment:
You presumably will want to revert moving this code when you fix the import cycle.

Reply:
done :)